backend-shared/util/operations/types_test.go

```diff
@@ -102,6 +103,188 @@ var _ = Describe("Testing CreateOperation function.", func() {
+				Expect(err).ToNot(HaveOccurred())
+				Expect(operationget.State).To(BeIdenticalTo(db.OperationState_Completed))
+
+				return true
+			}, "5s", "10ms").Should(BeTrue())
+		}
+
+		AfterEach(func() {
+			defer dbq.CloseDatabase()
+		})
+
+		It("should create a new DB Operation and test the waitForOperationToComplete function.", func() {
+			scheme,
+				argocdNamespace,
+				kubesystemNamespace,
+				workspace,
+				err := tests.GenericTestSetup()
+			Expect(err).ToNot(HaveOccurred())
+
+			err = db.SetupForTestingDBGinkgo()
+			Expect(err).ToNot(HaveOccurred())
+
+			ctx = context.Background()
+			log := log.FromContext(ctx)
+
+			k8sClientOuter := fake.NewClientBuilder().
+				WithScheme(scheme).
+				WithObjects(workspace, argocdNamespace, kubesystemNamespace).
+				Build()
+
+			k8sClient := &sharedutil.ProxyClient{
+				InnerClient: k8sClientOuter,
+			}
+
+			dbq, err = db.NewUnsafePostgresDBQueries(true, true)
+			Expect(err).ToNot(HaveOccurred())
+
+			_, _, _, gitopsEngineInstance, _, err := db.CreateSampleData(dbq)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("define a sample operation")
+			sampleOperation := &db.Operation{
+				Operation_id:            "test-engine-operation",
+				Instance_id:             gitopsEngineInstance.Gitopsengineinstance_id,
+				Resource_id:             "test-fake-resource-id",
+				Resource_type:           "GitopsEngineInstance",
+				Operation_owner_user_id: "test-user",
+			}
+
+			// Create the Operation which replaces the CreateOperation step in types' CreateOperation function
+			err = dbq.CreateOperation(ctx, sampleOperation, sampleOperation.Operation_owner_user_id)
+			Expect(err).ToNot(HaveOccurred())
+
+			go func() {
+				delayUpdateOfState(sampleOperation, k8sClient, *gitopsEngineInstance)
+			}()
+
+			// Now directly call waitForOperationToComplete which calls isOperationComplete to test these functions
+			err = waitForOperationToComplete(ctx, sampleOperation, dbq, log)
+			Expect(err).ToNot(HaveOccurred())
+
+			rowsAffected, err := dbq.DeleteOperationById(ctx, sampleOperation.Operation_id)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(rowsAffected).Should(Equal(1))
+
+		})
+
+		It("It should test CreateOperation with waitForOperation set to true to indirectly test the waitForOperationToComplete function.", func() {
+			scheme,
+				argocdNamespace,
+				kubesystemNamespace,
+				workspace,
+				err := tests.GenericTestSetup()
+			Expect(err).ToNot(HaveOccurred())
+
+			err = db.SetupForTestingDBGinkgo()
+			Expect(err).ToNot(HaveOccurred())
+
+			ctx = context.Background()
+			log := log.FromContext(ctx)
+
+			k8sClientOuter := fake.NewClientBuilder().
+				WithScheme(scheme).
+				WithObjects(workspace, argocdNamespace, kubesystemNamespace).
+				Build()
+
+			k8sClient := &sharedutil.ProxyClient{
+				InnerClient: k8sClientOuter,
+			}
+
+			dbq, err = db.NewUnsafePostgresDBQueries(true, true)
+			Expect(err).ToNot(HaveOccurred())
```

Comment:
can move to BeforeEach block (203 - 226)

Reply:
Done too.